### PR TITLE
Feature #67 Game Center login for Profile

### PR DESCRIPTION
**Pull request by EugeneButusov**

### CODE REVIEW

File: Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs [hunk 1]
```diff
@@ -371,6 +378,18 @@ void ApplyIntegretionLibraries (string socialPlatform, bool remove)
 
 				EditorGUI.EndDisabledGroup();
 				break;
+			case "gameCenter":
+#if UNITY_IOS
```
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs]: @EugeneButusov, are you sure you need `#if` here? What if an user has Android active at the moment, but he creates an app for both platform? (It's related to other cases in this file as well.)

File: Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs [hunk 2]
```diff
@@ -80,48 +80,48 @@ protected UserProfile(Provider provider, string profileId, string username)
 			this.Username = JSONObject.DecodeJsString(jsonUP[PJSONConsts.UP_USERNAME].str);
 			this.ProfileId = JSONObject.DecodeJsString(jsonUP[PJSONConsts.UP_PROFILEID].str);
 
-			if (jsonUP[PJSONConsts.UP_FIRSTNAME]) {
```
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @EugeneButusov, why did you remove the previous checking?

File: Soomla/Assets/WebPlayerTemplates/SoomlaConfig/ios/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../../../submodules/ios-profile/build/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a
```
Comment (vedi) [inline Soomla/Assets/WebPlayerTemplates/SoomlaConfig/ios/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a:1]: @EugeneButusov, will it work well in pre-built bundles?

File: Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs (hunk 1 above)
Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs]: @vedi, I'm not completely sure, but add this to avoid user confusing when he will work with Android and can be impressed how we implemented GameCenter for Android (if shown, then available).

File: Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs (hunk 2 above)
Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @vedi, because JSONObject can be not-null, but contains NULL-string (returned by `.str` method). Might be better is to add `jsonUP[blahblah].str != null`, but I'm thinking that this method is more versatile.

Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @EugeneButusov, I got this point, but I mean another... What if `jsonUP[PJSONConsts.UP_FIRSTNAME]` is `null`? Is it possible? 
It's not ObjC world :)

Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @vedi, looks like I understand. Do you mean to make this:
`jsonUP[PJSONConsts.UP_FIRSTNAME] != null && jsonUP[PJSONConsts.UP_FIRSTNAME].type != JSONObject.Type.NULL`
?

> What if jsonUP[PJSONConsts.UP_FIRSTNAME] is null? Is it possible? 

It may happen for non-native things or for future providers.

Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @EugeneButusov, yes, we're on the same page. Or you can just add your new condition to the old one.

File: Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs (hunk 1 above)
Comment (vedi) [inline Soomla/Assets/Plugins/Soomla/Profile/Config/ProfileSettings.cs]: @EugeneButusov, we have similar precendents in SoomlaStore, where iOS and Android params are visible for all platforms. I think it's less confusing, than magically disappearing settings, when a developer changes target platform before building. Unlike cocos2d-x, Unity3d projects are the same for all platforms, and we need adapt to this.

File: Soomla/Assets/WebPlayerTemplates/SoomlaConfig/ios/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a (hunk 3 above)
Comment (EugeneButusov) [inline Soomla/Assets/WebPlayerTemplates/SoomlaConfig/ios/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a:1]: @vedi, please describe what do you mean?

Comment (vedi) [inline Soomla/Assets/WebPlayerTemplates/SoomlaConfig/ios/ios-profile-gamecenter/libSoomlaiOSProfileGameCenter.a:1]: @EugeneButusov, I have a concern if you use such path, it can be put to the release bundles, but in those bundles, there is no file at this path.

File: Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs (hunk 2 above)
Comment (EugeneButusov) [inline Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs:83]: @vedi, fixed.
